.github/workflows/ci.yml

```diff
@@ -61,20 +64,19 @@ jobs:
           build: '${{ matrix.projectName }}'
       - name: 'Lint'
         run: 'pnpm --filter="${{ matrix.projectName }}" ${{ matrix.command }}'
-  project-test-jobs:
+
+  project-default-test-jobs:
     name: 'Test - ${{ matrix.projectName }} - ${{ matrix.name }}'
     runs-on: 'ubuntu-20.04'
     needs: 'project-jobs'
-    if: ${{ needs.project-jobs.outputs.test-jobs != '[]' }}
+    if: ${{ needs.project-jobs.outputs.default-test-jobs != '[]' }}
     strategy:
       fail-fast: false
       matrix:
-        include: ${{ fromJSON( needs.project-jobs.outputs.test-jobs ) }}
+        include: ${{ fromJSON( needs.project-jobs.outputs.default-test-jobs ) }}
     steps:
-      - uses: 'actions/checkout@v3'
+      - uses: 'actions/checkout@v4'
         name: 'Checkout'
-        with:
-          fetch-depth: 0
```

Comment:
Got it.
So, `fetch-depth: 0` makes sense for `project-jobs` and `project-lint-jobs`, but it's not required for test jobs, correct? I would like to optimize on this if possible, fetching the entire history add unnecessary execution time.

Reply:
@ObliviousHarmony can you confirm that testing jobs do not need `fetch-depth: 0`?

Reply:
To be honest, my biggest focus in optimization here was in making the project job generation step as fast as possible. At present I _think_ it's unnecessary, however, it seems reasonable that at some point it might be needed. At the time such a thing happens, it won't actually be very obvious to someone who isn't aware of the behavior of `actions/checkout` why it doesn't work.

Coming back to the `lint` question, if `fetch-depth` being unset is a significant enough improvement to warrant it, what if the matrix gave an option for it? The _very small_ use-case where it's needed could set it while the rest wouldn't. Some sort of `"optimizations"` config option for `lint` and `test` jobs?

```json
{
  "optimizations": {
    "fetchDepth": false
  }
}
```

This would cause `fetch-depth: 0` to be set. It would only be needed on Woo's lint job right now.

Reply:
For `lint` jobs I've added the `fetch-depth` back. The `optimizations` config sure sounds like an interesting approach if we'll ever need to optimize here.